packages/react-core/src/components/SearchInput/SearchInput.tsx

```diff
@@ -202,7 +214,15 @@ const SearchInputBase: React.FunctionComponent<SearchInputProps> = ({
     }
   };
 
-  const buildSearchTextInputGroup = ({ ...searchInputProps } = {}) => (
+  const onExpandHandler = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    setSearchValue('');
+    onToggleExpand(isExpanded, event);
+  };
```

Comment:
Not blocking and more of a followup issue since it'll depend whether the behavior should be built-in or just part of the example, but we should tidy up the focus management when the search input in the new example expands and collapses. Currently the behavior makes it difficult to know what has focus after interaction:

https://user-images.githubusercontent.com/70952936/189958482-15a604dc-d2f6-4837-abce-b38fb10a0b12.mov